File: src/main/java/com/example/demo/moongproject/model/NoticeModel.java
package com.example.demo.moongproject.model;

import java.util.List;

/**
 * 게시판(공지사항) Model
 */
public class NoticeModel {
  private TbNotice tbNotice;
  private List<TbNotice> tbNoticeList;

  public TbNotice getTbNotice() {
    return this.tbNotice;
  }

  public void setTbNotice(TbNotice tbNotice) {
    this.tbNotice = tbNotice;
  }
  
  public List<TbNotice> getTbNoticeList() {
	return this.tbNoticeList;
  }

  public void setTbNoticeList(List<TbNotice> tbNoticeList) {
	this.tbNoticeList = tbNoticeList;
  }
  
}

File: frontend/moongproject/src/application/board/model/NoticeModel.ts
import TbNotice from "domain/notice/TbNotice";
import { action, observable } from "mobx";

/**
 * 게시판(공지사항) Model
 */
export default class NoticeModel  {
  @observable private tbNotice: TbNotice;
  @observable private tbNoticeList: TbNotice[];

  public constructor(data?) {

    this.tbNotice = new TbNotice(data?.tbNotice || {});
  }

  public getTbNotice(): TbNotice {
    return this.tbNotice;
  }

  @action
  public setTbNotice(tbNotice: TbNotice): void {
    this.tbNotice = tbNotice;
  }

  public getTbNoticeList(): TbNotice[] {
    return this.tbNoticeList;
  }

  @action
  public setTbNoticeList(tbNoticeList: TbNotice[]): void {
    this.tbNoticeList = tbNoticeList;
  }
}
